numba/core/extending.py

```diff
@@ -217,19 +235,18 @@ def take_impl(arr, indices):
                     return res
                 return take_impl
     """
-    from numba.core.typing.templates import make_overload_method_template
+    return _overload_method_common(typ, attr, **kwargs)
 
-    def decorate(overload_func):
-        template = make_overload_method_template(
-            typ, attr, overload_func,
-            inline=kwargs.get('inline', 'never'),
-            prefer_literal=kwargs.get('prefer_literal', False)
-        )
-        infer_getattr(template)
-        overload(overload_func, **kwargs)(overload_func)
-        return overload_func
 
-    return decorate
+def overload_classmethod(typ, attr, **kwargs):
+    """
+    A decorator marking the decorated function as typing and implementing
+    classmethod *attr* for the given Numba type in nopython mode.
+
+
+    Similar to ``overload_method``.
+    """
```

Comment:
Should add this (and an example) to the high-level extending docs?

Reply:
Added docs and coverted `overload_method` and `overload_classmethod` to use sphinx autofunction in the high-level extending docs.

Reply:
Thanks, I've checked the rendering locally.